src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java

```diff
@@ -0,0 +1,262 @@
+    @In
+    private DelayManager delayManager;
+
+    private FastRandom random = new FastRandom();
+
+    /**
+     * Called when a bush seed is planted
+     *
+     * @param event         The seed planting event
+     * @param bush          The bush entity that has just been made
+     * @param bushComponent The bush definition component
+     */
+    @ReceiveEvent
+    public void onBushPlanted(OnSeedPlanted event, EntityRef bush, BushDefinitionComponent bushComponent) {
+        bushComponent.position = event.getPosition();
+        bushComponent.currentStage = -1;
+
+        EntityRef newBush = doBushGrowth(bushComponent, 1);
+        resetDelay(newBush, bushComponent.stages[0].minTime, bushComponent.stages[0].maxTime);
+        bush.saveComponent(bushComponent);
+    }
+
+    @ReceiveEvent
+    public void onBushComponent(OnActivatedComponent event, EntityRef bush, BushDefinitionComponent bushComponent) {
+        bushComponent.stages = buildGrowthStages(bushComponent.growthStages);
+        bush.saveComponent(bushComponent);
+    }
+
+    /**
+     * Called when a bush is destroyed.
+     * Handles the case of the bush actually being a vine bud.
+     *
+     * @param event  The destroy event
+     * @param entity The bush entity or a dummy entity if a bud.
+     */
+    @ReceiveEvent
+    public void onBushDestroyed(DoDestroyPlant event, EntityRef entity, BushDefinitionComponent bushComponent) {
+        int numSeeds = 1;
+        if (bushComponent.parent == null) {
+            /* It is a bush being destroyed */
+            if (bushComponent.currentStage == bushComponent.stages.length - 1) {
+                numSeeds = random.nextInt(1, 3);
+            }
+        } else {
+            /* It is a bud being destroyed */
+            if (!event.isParentDead) {
+                bushComponent.parent.send(new DoRemoveBud());
+            }
+            worldProvider.setBlock(bushComponent.position, blockManager.getBlock(BlockManager.AIR_ID));
+        }
+
+        /* Drop some seeds */
+        for (int i = 0; i < numSeeds; i++) {
+            Prefab seed = bushComponent.seed == null ? bushComponent.produce : bushComponent.seed;
+            EntityRef seedItem = entityManager.create(seed);
+            Vector3f position = bushComponent.position.toVector3f().add(0, 0.5f, 0);
+            seedItem.send(new DropItemEvent(position));
+            seedItem.send(new ImpulseEvent(random.nextVector3f(30.0f)));
+        }
+
+
+    }
+
+    /**
+     * Called when the bush should grow
+     *
+     * @param event         The event indicating the timer has ended
+     * @param bush          The bush to grow
+     * @param bushComponent The bush's definition
+     */
+    @ReceiveEvent
+    public void onBushGrowth(DelayedActionTriggeredEvent event, EntityRef bush, BushDefinitionComponent bushComponent) {
+        EntityRef newBush = null;
+        if (bushComponent.currentStage < bushComponent.stages.length - 1) {
```

Comment:
good thing we have `isInLastStage(bushComponent)`, so we can replace this with `!isInLastStage(bushComponent)`